api/handler/locking.go

```diff
@@ -287,15 +287,15 @@ func checkLockConfiguration(conf *data.ObjectLockConfiguration) error {
 
 	retention := conf.Rule.DefaultRetention
 	if retention.Mode != governanceMode && retention.Mode != complianceMode {
-		return fmt.Errorf("invalid Mode value: %s", retention.Mode)
+		return s3errors.GetAPIError(s3errors.ErrMalformedXML)
```

Comment:
Why not `GetAPIErrorWithError` to give a bit more details to the caller?

Reply:
Updated